packages/headless/src/api/search/search/product-recommendation.ts

```diff
@@ -1,6 +1,6 @@
 export interface ProductRecommendation {
   /**
-   * Document Uri in the index. Useful for analytics.
+   * @deprecated This parameter is no longer used by the Coveo platform. There is no need to provide it.
    */
   documentUri: string;
```

Comment:
Despite the deprecation warning, the parameter is still marked as required, which makes the warning unsolvable.

Also, it looks like `ProductRecommendation` is generally used as a return type, not an input type (the exception being some analytics).

I think that either:
1. The analytics calls should use a separate interface as an input, where `documentUri` is optional.
2. `documentUri` should be optional in this interface. This would in theory be breaking if somebody is reading it, but I don't expect that a whole lot of people would do that.

In both cases above, I think the input interface shouldn't say "There is no need to provide it".